src/main/java/formflow/library/UploadController.java

```diff
@@ -61,6 +65,21 @@ public ResponseEntity<?> upload(
         httpSession.setAttribute("id", submission.getId());
       }
       String fileExtension = Files.getFileExtension(Objects.requireNonNull(file.getOriginalFilename()));
+
+      if (fileExtension.equals("pdf")) {
+        try (PDDocument pdfFile = PDDocument.load(file.getInputStream())) {
+        } catch (InvalidPasswordException e) {
+          // TODO update when we add internationalization to use locale for message source
+          String message = localeLibraryConfiguration.messageSource()
+              .getMessage("upload-documents.error-password-protected", null, null);
+          return new ResponseEntity<>(message, HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
I wonder if a 415 is more appropriate - it's an unsupported media type, though the type itself is supported.

Reply:
Never mind, 422 is fine.